File: repositories/user_repository.py
class UserRepository:
    def __init__(self):
        self.users = {}

    def create_user(self, username, hashed_password, email):
        self.users[username] = {
            'username': username,
            'password': hashed_password,
            'email': email
        }

    def get_user_by_username(self, username):
        return self.users.get(username)

File: backend/repositories/users/userRepository.js
const db = require('../../config/db');

const createUser = async ({ email, password }) => {
  const result = await db.query('INSERT INTO users (email, password) VALUES (?, ?)', [email, password]);
  return result.insertId ? { id: result.insertId, email } : null;
};

const findUserByEmail = async (email) => {
  const result = await db.query('SELECT * FROM users WHERE email = ?', [email]);
  return result[0];
};

module.exports = {
  createUser,
  findUserByEmail,
};